File: 1750-minimum-length-of-string-after-deleting-similar-ends/1750-minimum-length-of-string-after-deleting-similar-ends.java
class Solution {
    public int minimumLength(String s) {
        int lo = 0, hi = s.length() - 1;

        while (lo < hi && s.charAt(lo) == s.charAt(hi)) {
            char target = s.charAt(lo);
            while (lo <= hi && s.charAt(lo) == target) {
                lo++;
            }
            while (lo <= hi && s.charAt(hi) == target) {
                hi--;
            }
        }

        return hi - lo + 1;
    }
}

File: 1750-minimum-length-of-string-after-deleting-similar-ends/1750-minimum-length-of-string-after-deleting-similar-ends.py
class Solution:
    def minimumLength(self, s: str) -> int:
        lo, hi = 0, len(s) - 1

        while lo < hi and s[lo] == s[hi]:
            target = s[lo]
            while lo <= hi and s[lo] == target:
                lo += 1
            while lo <= hi and s[hi] == target:
                hi -= 1

        return hi - lo  + 1